modules/web-mercator/src/get-bounds.js

```diff
@@ -0,0 +1,44 @@
+import {worldToLngLat} from './web-mercator-utils';
+import * as vec2 from 'gl-matrix/vec2';
+import {transformVector} from './math-utils';
+
+const DEGREES_TO_RADIANS = Math.PI / 180;
+
+// Returns the quad at the intersection of the frustum and the goven z plane
+export default function getBounds(viewport, z = 0) {
+  const {width, height, unproject} = viewport;
+  const unprojectOps = {targetZ: z};
+  const bottomLeft = unproject([0, height], unprojectOps);
+  const bottomRight = unproject([width, height], unprojectOps);
+  let topLeft;
+  let topRight;
+
+  const halfFov = Math.atan(0.5 / viewport.altitude);
+  const angleToGround = (90 - viewport.pitch) * DEGREES_TO_RADIANS;
+  // The top plane is parallel to the ground if halfFov == angleToGround
+  if (halfFov > angleToGround - 0.01) {
+    // intersect with the far plane
+    topLeft = unprojectOnFarPlane(viewport, 0, z);
+    topRight = unprojectOnFarPlane(viewport, width, z);
+  } else {
+    // intersect with the top plane
+    topLeft = unproject([0, 0], unprojectOps);
+    topRight = unproject([width, 0], unprojectOps);
+  }
+
+  return [bottomLeft, bottomRight, topRight, topLeft];
+}
+
+function unprojectOnFarPlane(viewport, x, targetZ) {
```

Comment:
Maybe a comment header? What does "far plane" refer to here? It gives associations to the "far clipping plane".